src/components/form.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react';
+//import axios from 'axios';
+import { Divider,Card, Form, Input, Button,Label} from 'semantic-ui-react';
+import {
+  FormBuilder,
+  FieldGroup,
+  FieldControl,
+  Validators
+} from 'react-reactive-form'
+ 
+export default class Login extends Component {
+  // Initialize the empty group control
+  loginForm = FormBuilder.group({})
+ 
+  handleSubmit = e => {
+    console.log('Form values', this.loginForm.value)
+    e.preventDefault()
+  
+  }
+  render() {
+    return (
+        <Card>
+          <Card.Content>
+            <Card.Header>Sign Up Form</Card.Header>
+         </Card.Content>
+         <Divider />
+      <FieldGroup
+        control={this.loginForm}
+        render={({ invalid }) => (
+          <Form onSubmit={this.handleSubmit}>
+            <FieldControl
+              name="firstname"
+              options={{ validators: Validators.required }}
+              render={({ handler, touched, hasError}) => (
+                <div>
+                  <Label>First Name :</Label>
+                  <Input placeholder={'* Enter Your First Name'} {...handler()} />
+                  <span>
+                    {touched && hasError('required') && 'Username is required'}
+                  </span>
+                </div>
+              )}
+            />
+            <Divider />
+            <FieldControl
```

Comment:
https://react.semantic-ui.com/collections/form/#shorthand-field-control-id

use this for validations of all the fields

Reply:
have a look at this as well

https://react.semantic-ui.com/collections/form/#states-field-error-label